Signal/src/AppDelegate.m

```diff
@@ -102,6 +103,18 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
     DDLogWarn(@"%@ application: didFinishLaunchingWithOptions.", self.logTag);
 
+    // We need to this _after_ we set up logging but _before_ we do
+    // anything else.
+    [self ensureIsReadyForAppExtensions];
+
+#if RELEASE
+    // ensureIsReadyForAppExtensions may have changed the state of the logging
```

Comment:
I don't understand how ensureIsReadyForAppExtensions would change the state of the logging preference. Can you expound?

Reply:
I've elaborated the comment. It's because of `[NSUserDefaults migrateToSharedUserDefaults]`.